sdl2promela/stop_condition/translator.py

```diff
@@ -708,13 +775,120 @@ def _generate_queue_length_call(context: GenerateContext, expr: model.CallExpres
     )
 
 
+def _generate_get_state_for_toplevel(
+    context: GenerateContext, process_reference: model.VariableReference
+):
+    process_name = process_reference.name.lower()
+    if process_name not in context.processes:
+        raise TranslateException(
+            "Cannot find process with name '{}'".format(process_name)
+        )
+    context.process_state_selection = process_name
+    return (
+        promelaBuilder.MemberAccessBuilder()
+        .withUtypeReference(
+            promelaBuilder.MemberAccessBuilder()
+            .withUtypeReference(
+                promelaBuilder.VariableReferenceBuilder("global_state").build()
+            )
+            .withMember(promelaBuilder.VariableReferenceBuilder(process_name).build())
+            .build()
+        )
+        .withMember(promelaBuilder.VariableReferenceBuilder("state").build())
+        .build()
+    )
+
+
+def _generate_get_state_for_parallel(
+    context: GenerateContext, state_reference: model.Selector
+):
+    if not isinstance(state_reference.elements[0], model.VariableReference):
+        raise TranslateException("Invalid parameter for function 'get_state'")
+
+    process_name = typing.cast(
+        model.VariableReference, state_reference.elements[0]
+    ).name
+    if process_name not in context.processes:
+        raise TranslateException(
+            "Cannot find process with name '{}'".format(process_name)
+        )
+
+    process = context.processes[process_name]
+    aggregates = context.aggregates[process_name]
+    composites = process.composite_states
+    state_name = ""
+    state_prefix = ""
+
+    # Construct name of the state variable
+
+    for index in range(1, len(state_reference.elements)):
+        element_name = typing.cast(
+            model.VariableReference, state_reference.elements[index]
+        ).name.lower()
+
+        if len(state_name) == 0:
+            state_name = element_name
+            state_prefix = element_name
+        else:
+            state_name = state_name + SEPARATOR + element_name
+            state_prefix = state_prefix + "_" + element_name
+
+        candidates = [
```

Comment:
I think this (along with the following lines) would be more readable in the future if a function was extracted -> getStateOfName, returning Optional. What do you think?

Reply:
refactored, the same algo was used in two places, now it is a function